### PR TITLE
ci: Add package verifications to build workflow

### PR DESCRIPTION
**Pull request by eculver**
### Description
This is the last PR in the series to artifacts as part of the build workflow. The primary target for this last PR is for validating rpm and deb packages. There were TODOs for these last two jobs, which I'm adding. However, I'm also going back and making a few changes to the previous two PRs (#13200 and #13221):
- Verifications now all check the contents of the version string (as opposed to just running `consul version`)
- Renaming scripts to all match `verify_*.sh` pattern
- Moving zip/binary verification into its own job

### Testing & Reproduction steps
The output can be seen in this run: 

### Links
- [CTIA-14](https://hashicorp.atlassian.net/browse/CTIA-14)
- [CTIA-17](https://hashicorp.atlassian.net/browse/CTIA-17)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -283,90 +265,193 @@ jobs:
...
-        if: ${{ matrix.arch == 'amd64' }}
-        run: |
-          unzip ${{ env.PKG_NAME }}_${{env.version}}_darwin_${{ matrix.arch }}.zip
-          ./consul version
+          name: ${{ env.zip_name }}
+
+      - name: Unzip ${{ matrix.arch }} linux zip
+        run: unzip ${{ env.zip_name }}
+
+      - name: Run verification for ${{ matrix.arch }} linux binary
```
Comment (huikang) [inline .github/workflows/build.yml:341]: nit: darwin binary

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -283,90 +265,193 @@ jobs:
...
       fail-fast: true
+    env:
+      version: ${{ needs.get-product-version.outputs.product-version }}
+      deb_name: consul_${{ needs.get-product-version.outputs.product-version }}-*-${{ matrix.arch }}.deb
 
     name: Verify ${{ matrix.arch }} debian package
     steps:
       - uses: actions/checkout@v2
-
+
```
Comment (huikang) [inline .github/workflows/build.yml:361]: nit: remove the extra tab

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -283,90 +265,193 @@ jobs:
...
+      - name: Verify ${{ matrix.arch }} debian package (using Docker linux/arm/v7)
+        if: ${{ matrix.arch == 'arm' }}
         run: |
-          echo "TODO: run verification for ${{ matrix.arch }} debian package"
+          docker run \
+            --platform=linux/arm/v7 \
+            -v $(pwd):/workdir \
+            -w /workdir \
+            arm64v8/debian \
```
Comment (huikang) [inline .github/workflows/build.yml:385]: Should this be an `arm32v7/debian` image?

**Review by huikang — COMMENTED** [review]
Some minor comments

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -6,6 +6,7 @@ on:
     branches:
       # Push events on the main branch
       - main
+      - CTIA-17-verify-linux-packages
```
Comment (huikang) [inline .github/workflows/build.yml:9]: Let's make sure this line is removed before merging

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -276,90 +258,230 @@ jobs:
...
         with:
-          name: ${{ env.PKG_NAME }}_${{env.version}}_darwin_${{ matrix.arch }}.zip
-      - name: Run verification for ${{ matrix.arch }} linux zip
-        if: ${{ matrix.arch == 'amd64' }}
-        run: |
-          unzip ${{ env.PKG_NAME }}_${{env.version}}_darwin_${{ matrix.arch }}.zip
-          ./consul version
+          name: ${{ env.zip_name }}
+
+      - name: Unzip ${{ matrix.arch }} darwin zip
```
Comment (huikang) [inline .github/workflows/build.yml:331]: Since verify-darwin workflow doesn't have the arch matrix, not sure if it will cause any issue in these steps.

**Review by huikang — APPROVED** [review]

File: .github/workflows/build.yml (hunk 5 above)
Comment (eculver) [reply .github/workflows/build.yml:314]: Good call. I'll update it to use a static string.

**Pull request merged**